guardian/s3-upload
# ci: Adopt GitHub Actions for CI

## What does this change?
This change adopt GitHub Actions for CI, replacing TeamCity which DevX are switching off in Q2 2023.

This change isn't exactly a no-op switch, as the [`sbt-riffraff-artifact` plugin](https://github.com/guardian/sbt-riffraff-artifact) has been removed in favour of [`guardian/actions-riff-raff`](https://github.com/guardian/actions-riff-raff). Additionally, the `ci/build.sh` script has been removed; the steps in this file were simple enough to perform inline in the workflow YAML.

## How to test
[Deploy to CODE](https://riffraff.gutools.co.uk/deployment/view/b4f90ac3-00ed-40a7-9b77-45a2aad5d23f), and observe it succeeding, i.e. a file can be uploaded.

## Notes
- I've paused the [TeamCity project](https://teamcity.gutools.co.uk/buildConfiguration/EditorialTools_S3Uploader?mode=branches).
- The RIff-Raff project name of `media-service::teamcity::s3-uploader` has been retained, however it might be a little confusing now!
- Using https://github.com/guardian/riff-raff-checksum to demonstrate these changes are a no-op has failed. I think this is due to the removal of the SBT plugin. This is an indication that this change should be decoupled from the migration part, however the testing has given me confidence in the combined changes.
  - The files from [`main` are here](https://riffraff.gutools.co.uk/deployment/request/deployFiles?projectName=media-service%3A%3Ateamcity%3A%3As3-uploader&id=157), and the files from [this branch are here](https://riffraff.gutools.co.uk/deployment/request/deployFiles?projectName=media-service%3A%3Ateamcity%3A%3As3-uploader&id=167). We can see the _names_ are the same, however the file size differs, implying the files are not exactly the same.

### Changes in build time
TeamCity doesn't offer a complete historic view for this project. However the below graph suggests builds in TeamCity are ~1 minute:

![image](https://github.com/guardian/s3-upload/assets/836140/5c42668d-4950-4e16-8e1b-c0caa6b20107)

GitHub Actions is slightly over at ~1m45s.
